### PR TITLE
Windows devices  input fix

### PR DESCRIPTION
**Pull request by LorenzoDezi**
We have found an issue with Windows devices, where some touch events like TOUCH_MOVE were not registering. We resolve the issue using [pointer events](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events), which is the most modern API for events, addressing the need for a generic pointer that can handle different input devices.

### CODE REVIEW

File: pal/input/web/mouse-input.ts [hunk 1]
```diff
@@ -93,17 +93,18 @@ export class MouseInputSource {
...
         // register mouse move event
-        this._canvas?.addEventListener('mousemove', this._handleMouseMove);
+        this._canvas?.addEventListener('pointermove', this._createCallback(InputEventType.MOUSE_MOVE));
 
         // register mouse up event
-        window.addEventListener('mouseup', this._handleMouseUp);
-        this._canvas?.addEventListener('mouseup', this._handleMouseUp);
+        const handleMouseUp = this._createCallback(InputEventType.MOUSE_UP);
+        window.addEventListener('pointerup', handleMouseUp);
+        this._canvas?.addEventListener('pointerup', handleMouseUp);
```
Comment (PPpro) [inline pal/input/web/mouse-input.ts]: we have created callbacks in the constructor, why should we create again here ?

File: pal/input/web/touch-input.ts [hunk 2]
```diff
@@ -53,35 +53,26 @@ export class TouchInputSource {
...
+            if (touch) {
                 if (eventType === InputEventType.TOUCH_END || eventType === InputEventType.TOUCH_CANCEL) {
-                    touchManager.releaseTouch(touchID);
+                    touchManager.releaseTouch(event.pointerId);
                 }
                 handleTouches.push(touch);
             }
             event.stopPropagation();
-            if (event.target === this._canvas) {
+            if (event.target === this._canvas || InputEventType.TOUCH_MOVE) {
```
Comment (PPpro) [inline pal/input/web/touch-input.ts]: Is this `InputEventType.TOUCH_MOVE` condition statement neccessary ?

File: pal/input/web/mouse-input.ts (hunk 1 above)
Comment (PPpro) [reply pal/input/web/mouse-input.ts]: and can you fix the return type  of `_createCallback` method to `(pointerEvent: PointerEvent) => void`

File: pal/input/web/touch-input.ts [hunk 3]
```diff
@@ -53,35 +53,26 @@ export class TouchInputSource {
...
-                }
+            const location = this._getLocation(event, canvasRect);
+            const touch = touchManager.getTouch(event.pointerId, location.x, location.y);
+            if (touch) {
                 if (eventType === InputEventType.TOUCH_END || eventType === InputEventType.TOUCH_CANCEL) {
-                    touchManager.releaseTouch(touchID);
+                    touchManager.releaseTouch(event.pointerId);
                 }
                 handleTouches.push(touch);
             }
```
Comment (PPpro) [inline pal/input/web/touch-input.ts:73]: can pointer event support multi touch well ?

File: pal/input/web/touch-input.ts (hunk 2 above)
Comment (LorenzoDezi) [reply pal/input/web/touch-input.ts]: No, I fixed it in the new commit.

File: pal/input/web/touch-input.ts (hunk 3 above)
Comment (LorenzoDezi) [reply pal/input/web/touch-input.ts:73]: Yes, I tested it and it works.

**Comment by LorenzoDezi** [comment]
@PPpro there is also another fix needed. In order for the pointer events to work correctly along with these changes, we found out that the document.documentElement.style.touchAction property must be set to 'none'. In our projects, we use a custom component to make sure that property gets set correctly at startup since we couldn't find a way to set it directly from the cocos engine. Do you know how to generate the aforementioned style just with a custom modification directly in the cocos engine?

File: pal/input/web/mouse-input.ts (hunk 1 above)
Comment (LorenzoDezi) [reply pal/input/web/mouse-input.ts]: I fixed it in the new commit.

**Review by PPpro — APPROVED** [review]

**Comment by minggo** [comment]
@LorenzoDezi could you please fix conflict? Thanks.

**Comment by pandamicro** [comment]
Fixed

**Review by minggo — APPROVED** [review]

**Pull request merged**